src/Common/Crypto/KeyPair.h

```diff
@@ -0,0 +1,169 @@
+#pragma once
+
+#include "config.h"
+
+#include <Common/Exception.h>
+
+#if USE_SSL
+#include <Common/OpenSSLHelpers.h>
+#include <openssl/evp.h>
+#include <openssl/pem.h>
+#include <openssl/rsa.h>
+#endif
+
+#include <memory>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int OPENSSL_ERROR;
+}
+
+#if USE_SSL
+
+class KeyPair
```

Comment:
This file and X509Certificate.h are survivors of Poco, right?

We can unfortunately not simply copy-paste the code from base/poco to Common/Crypto. Poco is under boost license, we are Apache. Therefore, the code (only the surviving files) need to stay in base/poco/Crypto ...

Reply:
Nevermind, both are rewritten from scratch. Its fine to move them out of Poco.